plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -1234,6 +1245,15 @@ public Answer createTemplateFromVolume(CopyCommand cmd) {
 
             details = "create template from volume exception: " + VmwareHelper.getExceptionMessage(e);
             return new CopyCmdAnswer(details);
+        } finally {
+            try {
+                if (volume.getVmName() == null && vmMo != null) {
+                    vmMo.detachAllDisks();
+                    vmMo.destroy();
```

Comment:
Ah nevermind, worker VM will be destroyed here. Should we explictly set workerVmMo above and use that to destroy in finally? To avoid any regressions?

Reply:
there's a check that volume is in fact detached  with this : ```volume.getVmName()``` hence, vmMo under such a scenario will only point to the worker VM.